rfcs/react-components/extension-packages.md

```diff
@@ -0,0 +1,285 @@
+
+before an extension component can be elevated to the core library. The extension package should be deprecated and replaced
+with a new package that follows the naming conventions and versioning of the core library.
+
+Example:
+
+1. Contributor creates `@fluentui-contrib/react-people-picker`
+2. Contributor requests the Fluent UI team for elevation
+3. Fluent UI team assigns a memeber to review the package
+4. Changes are proposed to `@fluentui-contrib/react-people-picker`
+5. Changes are implemented to `@fluentui-contrib/react-people-picker`
+6. Fluent UI team creates new package `@fluentui/react-people-picker@9.0.0`
+
+Once a package is elevated to the core library, the ownership should be shared between the Fluent UI team and original
+authors.
+
+Any packages with competing implementations cannot be elevated until the competing implementations converge.
+
+### Different breaking change cadence
+
+Breaking changes for extension components can occur more frequently than for core components. The proposal does not
+mean that breaking changes will be taken lightly, or that there will be a requirement to make breaking changes often.
+
+##### Pros
+
+- 👍 Bleeding edge improvements for partners
+- 👍 Specialized components might not have widespread use and can be upgraded in a few instances
+
+##### Cons
+
+- 👎 Cost of breaking changes has not changed
+- 👎 Just because we make breaking changes does not mean users will upgrade earlier
+- 👎 Need to handle urgent bugfixes for most past versions - need a clear support SLA
+
+### How to depend on core components
+
+All extension packages should begin with a peer dependency on `@fluentui/react-components`. This is the recommendation used
+by [material-ui](https://mui.com/material-ui/getting-started/faq/#v4-why-arent-my-components-rendering-correctly-in-production-builds)
+for OSS authors that want to extend the library. This recommendation is made to [reduce duplicates in bundle](https://github.com/facebook/react/issues/13346#issuecomment-515079922).
+
+We can look at a case by case basis if more concrete dependencies on the core library are needed. This policy should
+be enforced by automation that needs to be disabled in the cases where stricter dependencies are necessary.
+
+### Packages scope and hosting
+
+#### New package scope
+
+A new scope `@fluentui-contrib/` will be created for extension packages that will be a part of the Fluent umbrella.
+However these packages are not considered a part of the core library. These packages must follow the requirements
+mentioned earlier above.
+
+There is no obligation to publish under this scope. However the requirements we set will be only be minimal to
+ensure some consistency under the Fluent UI umbrella, which will build confidence for users who want to consume
+Fluent UI conformant packages.
+
+#### Self code hosting
+
+Code for `@fluentui/contrib` packages should be self hosted. The Fluent UI team will provide bootstraping utilities
```

Comment:
Will these bootstrapping utilities have an update path? If Fluent improves the pipelines/storybook/etc. can we ensure downstream bootstrappers can easily update to get new behavior also?